src/server_manager/model/account.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright 2018 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as cloud_povider from './cloud_provider';
+import {ManagedServer, RegionId, RegionMap} from './server';
+
+export interface Data {
+  id: string;
+  displayName: string;
+  provider: cloud_povider.Id;
+  credential: unknown;
+}
+
+export interface AccountModelFactory<T extends Account> {
+  createAccountModel(data: Data): Promise<T>;
+}
+
+// TODO: This is a clone of the ManagedServerRepository interface. We should try
+// to make this generic, or remove it if that's not possible.
+export interface Account {
```

Comment:
This should probably be DigitalOceanAccount. As we saw, accounts for different clouds are different.